common/services/reference-data.test.js

```diff
@@ -106,30 +106,102 @@ describe('Reference Service', function() {
   })
 
   describe('#getLocations()', function() {
-    context('when request returns 200', function() {
-      let response
-
-      beforeEach(async function() {
-        nock(API.BASE_URL)
-          .get('/reference/locations')
-          .query({ page: '1', per_page: '100' })
-          .reply(200, locationsPage1Serialized)
-
-        nock(API.BASE_URL)
-          .get('/reference/locations')
-          .query({ page: '2', per_page: '100' })
-          .reply(200, locationsPage2Serialized)
-
-        response = await getLocations()
+    context('with no parameters', function() {
+      context('when request returns 200', function() {
+        let response
+
+        beforeEach(async function() {
+          nock(API.BASE_URL)
+            .get('/reference/locations')
+            .query({ page: '1', per_page: '100' })
+            .reply(200, locationsPage1Serialized)
+
+          nock(API.BASE_URL)
+            .get('/reference/locations')
+            .query({ page: '2', per_page: '100' })
+            .reply(200, locationsPage2Serialized)
+
+          response = await getLocations()
+        })
+
+        it('should call API with no filter query parameters', function() {
+          expect(nock.isDone()).to.be.true
+        })
+
+        it('should return a full list of locations', function() {
+          expect(response.length).to.deep.equal(
+            locationsDeserialized.data.length
+          )
+          expect(response).to.deep.equal(locationsDeserialized.data)
+        })
       })
+    })
 
-      it('should call API', function() {
-        expect(nock.isDone()).to.be.true
+    context('with type parameter', function() {
+      context('when request returns 200', function() {
+        let type
+
+        beforeEach(async function() {
+          type = 'police'
+
+          nock(API.BASE_URL)
+            .get('/reference/locations')
+            .query({
+              page: '1',
+              per_page: '100',
+              filter: { location_type: type },
+            })
+            .reply(200, locationsPage1Serialized)
+
+          nock(API.BASE_URL)
+            .get('/reference/locations')
+            .query({
+              page: '2',
+              per_page: '100',
+              filter: { location_type: type },
+            })
+            .reply(200, locationsPage2Serialized)
+
+          await getLocations({ type: type })
+        })
+
+        it('should call API with the type filter query parameter', function() {
+          expect(nock.isDone()).to.be.true
+        })
       })
+    })
 
-      it('should return a full list of locations', function() {
-        expect(response.length).to.deep.equal(locationsDeserialized.data.length)
-        expect(response).to.deep.equal(locationsDeserialized.data)
+    context('with nomisAgencyId parameter', function() {
```

Comment:
I've been thinking about how we can write these tests a bit better and I think this highlights we may need a refactor.

I think we should test the serialization/deserialization in the API client and in these services we should move the apiClient response rather than using nock.

But one for later on anyway.

Reply:
Yes there’s definitely scope for de-duplication in these tests, but it’s one for another day.

Reply:
Yeah, also hoping that refactor could fix #82 too